numpy/random/tests/test_smoke.py

```diff
@@ -708,102 +781,127 @@ def test_integers_broadcast_errors(self, dtype):
-        cls.seed_vector_bits = 192
-        cls._extra_setup()
+    def _create_gen(cls):
+        bit_generator = SFC64
+        advance = None
+        seed = [12345]
+        rg = Generator(bit_generator(*seed))
+        initial_state = rg.bit_generator.state
+        seed_vector_bits = 192
+        seed_error = TypeError
+        return RNGData(
+            bit_generator, advance, seed, rg,
+            initial_state, seed_vector_bits, seed_error
+        )
 
 
 class TestPCG64(RNG):
     @classmethod
-    def setup_class(cls):
-        cls.bit_generator = PCG64
-        cls.advance = 2**63 + 2**31 + 2**15 + 1
-        cls.seed = [12345]
-        cls.rg = Generator(cls.bit_generator(*cls.seed))
-        cls.initial_state = cls.rg.bit_generator.state
-        cls.seed_vector_bits = 64
-        cls._extra_setup()
+    def _create_gen(cls):
+        bit_generator = PCG64
+        advance = 2**63 + 2**31 + 2**15 + 1
+        seed = [12345]
+        rg = Generator(bit_generator(*seed))
+        initial_state = rg.bit_generator.state
+        seed_vector_bits = 64
+        seed_error = TypeError
+        return RNGData(
+            bit_generator, advance, seed, rg,
+            initial_state, seed_vector_bits, seed_error
+        )
 
 
 class TestPCG64DXSM(RNG):
     @classmethod
-    def setup_class(cls):
-        cls.bit_generator = PCG64DXSM
-        cls.advance = 2**63 + 2**31 + 2**15 + 1
-        cls.seed = [12345]
-        cls.rg = Generator(cls.bit_generator(*cls.seed))
-        cls.initial_state = cls.rg.bit_generator.state
-        cls.seed_vector_bits = 64
-        cls._extra_setup()
+    def _create_gen(cls):
+        bit_generator = PCG64DXSM
+        advance = 2**63 + 2**31 + 2**15 + 1
+        seed = [12345]
+        rg = Generator(bit_generator(*seed))
+        initial_state = rg.bit_generator.state
+        seed_vector_bits = 64
+        seed_error = TypeError
+        return RNGData(
+            bit_generator, advance, seed, rg,
+            initial_state, seed_vector_bits, seed_error
+        )
 
 
 class TestDefaultRNG(RNG):
     @classmethod
-    def setup_class(cls):
+    def _create_gen(cls):
         # This will duplicate some tests that directly instantiate a fresh
         # Generator(), but that's okay.
-        cls.bit_generator = PCG64
-        cls.advance = 2**63 + 2**31 + 2**15 + 1
-        cls.seed = [12345]
-        cls.rg = np.random.default_rng(*cls.seed)
-        cls.initial_state = cls.rg.bit_generator.state
-        cls.seed_vector_bits = 64
-        cls._extra_setup()
+        bit_generator = PCG64
+        advance = 2**63 + 2**31 + 2**15 + 1
+        seed = [12345]
+        rg = np.random.default_rng(*seed)
+        initial_state = rg.bit_generator.state
+        seed_vector_bits = 64
+        seed_error = TypeError
+        return RNGData(
+            bit_generator, advance, seed, rg,
+            initial_state, seed_vector_bits, seed_error
+        )
```

Comment:
how about this instead:
```suggestion
        return RNGData(
            bit_generator = PCG64
            advance = 2**63 + 2**31 + 2**15 + 1
            seed = [12345]
            rg = np.random.default_rng(*seed)
            initial_state = rg.bit_generator.state
            seed_vector_bits = 64
            seed_error = TypeError
        )
```

You can make a similar edit for the other `RNGData` you create above.